### PR TITLE
Implement new iterator protocol

### PR DESCRIPTION
**Pull request by mkborregaard**

### CODE REVIEW

File: src/utils.jl [hunk 1]
```diff
@@ -215,15 +215,13 @@ anynan(i::Int, args::Tuple) = any(a -> try isnan(_cycle(a,i)) catch MethodError
 anynan(istart::Int, iend::Int, args::Tuple) = any(i -> anynan(i, args), istart:iend)
 allnan(istart::Int, iend::Int, args::Tuple) = all(i -> anynan(i, args), istart:iend)
 
-function Base.start(itr::SegmentsIterator)
-    nextidx = 1
-    if !any(isempty,itr.args) && anynan(1, itr.args)
-        _, nextidx = next(itr, 1)
+Base.iterate(itr::SegmentsIterator, nextidx::Int) =
```
Comment (daschw) [inline src/utils.jl]: This does not seem right.

**Comment by daschw** [comment]
Nice work today!!!
Don't you have a conference to attend?

**Comment by mkborregaard** [comment]
closing in favour of https://github.com/JuliaPlots/Plots.jl/pull/1630

**Comment by mkborregaard** [comment]
Worse - I've a talk to prepare - will do that now.

**Comment by daschw** [comment]
Good luck with that!

**Pull request closed without merging**